Building.md

```diff
@@ -4,7 +4,7 @@ To learn more about how to use CUDA Quantum, please take a look at the [CUDA
 Quantum Documentation][cuda_quantum_docs]. The page also contains [installation
 instructions][official_install] for released packages.
 
-[cuda_quantum_docs]: https://nvidia.github.io/cuda-quantum/
+[cuda_quantum_docs]: https://nvidia.github.io/cuda-quantum/latest
```

Comment:
I think it may make sense to direct the nvidia.github.io/cuda-quantum link to the docs for the latest stable release on NGC opposed to the docs for the latest main. However, if someone is looking at the GitHub source code, the docs version they want is latest. I am hence explicitly setting the link here to that version.